rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/model/PathConfigs.java

```diff
@@ -78,6 +79,41 @@ private PathConfig buildPathConfig(ISourceLocation projectRoot) {
         }
     }
 
+    private void recalcPathConfigIfChanged(ISourceLocation projectRoot, URIResolverRegistry reg, ISourceLocation targetFile)
+        throws IOException {
+        reg.watch(targetFile, false, changedManifest ->
+            currentPathConfigs.replace(projectRoot, actualBuild(projectRoot))
+        );
+    }
+
+    private void registerMavenWatches(URIResolverRegistry reg, ISourceLocation projectRoot) throws IOException {
+        var mainPom = URIUtil.getChildLocation(projectRoot, "pom.xml");
+        if (reg.exists(mainPom)) {
+            recalcPathConfigIfChanged(projectRoot, reg, mainPom);
+            if (hasParentSection(reg, mainPom)) {
+                var parentPom = URIUtil.getChildLocation(URIUtil.getParentLocation(projectRoot), "pom.xml");
```

Comment:
are pom parents always in immediate folder parents?

Reply:
In most cases, yes, but you are not required. I wanted to go for a heuristic that was good enough in most cases, as in reality it's hard to solve.

Reply:
I think that 99.9% of the people will not put newlines, but there are possibly many that might put spaces. So let's write a regex.